lib/adapters/tool/playwright/ipc.ts

```diff
@@ -0,0 +1,17 @@
+import _ from 'lodash';
+import type {ChildProcess} from 'node:child_process';
+
+export default {
+    emit: (event: string, data: Record<string, unknown> = {}): void => {
+        process.send && process.send({event, ...data});
+    },
+    on: <T extends Record<string, unknown>>(event: string, handler: (msg: T & {event: string}) => void, proc: ChildProcess | NodeJS.Process = process): void => {
```

Comment:
Hmm didn't we have more convenient tools for IPC somewhere? So that we didn't need to implement this from scratch. Not critical.

Reply:
I used a similar solution from testplane - https://github.com/gemini-testing/testplane/blob/8c82cd4468eddd7a2a7dafac119fd4f840fdacd4/src/utils/ipc.js. There are only ~10 lines of simple code here I don't think that we should find some library for this (but I searched on the Internet and didn't find simple solution)